witness/tests/Test/Dscp/Witness/Explorer/ExplorerSpec.hs

```diff
@@ -78,158 +100,247 @@ getTransactionsPaged chunkSize mAddress = getFrom Nothing
-            , property $ isNothing (wbiBlockInfo tx2)
-            ]
-
-    it "Filtering on address works fine (when the address is tx input)" $ witnessProperty $ do
-        let selectSecret = oneof [selectGenesisSecret, pure testSomeGenesisSecret]
-            interestingAddress = mkAddr $ toPublic testSomeGenesisSecret
-        n <- pick $ choose (1, 5)
-        txs <- replicateM n $ createAndSubmitTx selectSecret
-        _ <- lift dumpBlock
-
-        let expected = filter (\tx -> interestingAddress `elem` txRelatedAddrs tx) txs
-        res <- lift $ getTransactions Nothing Nothing (Just interestingAddress)
-        let resTop = reverse . take (length expected) $ plItems res
-        return $ map (toTxId . wbiItem) resTop
-                 ===
-                 map toTxId expected
-
-  describe "getPublications" $ do
-    it "Returns all transactions at once just fine" $ once $ witnessProperty $ do
-        n <- pick $ choose (1, 3)
-        txs <- replicateM n $ createAndSubmitPub (pure testSomeGenesisSecret)
-        _ <- lift dumpBlock
-
-        res <- lift $ getPublications Nothing Nothing Nothing
-        let resTop = reverse $ plItems res
-        return $ map (toPtxId . wbiItem) resTop
-                 ===
-                 map toPtxId txs
-
-    it "Filtering on author works fine" $ witnessProperty $ do
-        let selectSecret = oneof [selectGenesisSecret, pure testSomeGenesisSecret]
-            interestingAddress = mkAddr $ toPublic testSomeGenesisSecret
-        n <- pick $ choose (1, 5)
-        txs <- replicateM n $ createAndSubmitPub selectSecret
-        _ <- lift dumpBlock
-
-        let expected = filter (\tx -> interestingAddress == ptAuthor tx) txs
-        res <- lift $ getPublications Nothing Nothing (Just interestingAddress)
-        let resTop = reverse $ plItems res
-        return $ map (toPtxId . wbiItem) resTop
-                 ===
-                 map toPtxId expected
-
-    it "Blocks info is present" . once $ witnessProperty $ do
-        _ <- createAndSubmitPub selectGenesisSecret
-        blockHash <- lift dumpBlock
-        _ <- createAndSubmitPub selectGenesisSecret
-
-        res <- lift $ getPublications Nothing Nothing Nothing
-        let tx2 : tx1 : _ = plItems res
-        return $ conjoin
-            [ fmap biHeaderHash (wbiBlockInfo tx1) === Just blockHash
-            , property $ isNothing (wbiBlockInfo tx2)
-            ]
-
-  describe "getHashType" $ do
-    it "Handles (mempool) money transactions correctly" . once $ witnessProperty $ do
-        tx <- createAndSubmitTx selectGenesisSecret
-        res <- lift $ getHashType (toHex $ toTxId tx)
-        return $ res === HashIsMoneyTx
-
-    it "Handles (mempool) pub transactions correctly" . once $ witnessProperty $ do
-        tx <- createAndSubmitPub selectGenesisSecret
-        res <- lift $ getHashType (toHex $ toPtxId tx)
-        return $ res === HashIsPublicationTx
+    describe "getTransaction" $ do
```

Comment:
@Martoon-00 Setting up different indentation rules for different files in the project seems like a shitty idea, because it means complicating editor settings for everyone. There are only few specs which use two-spaces indentation, and frankly, I feel ashamed that I apparently missed it during some previous reviews.

Reply:
Meh, okay